### PR TITLE
bump kind job to kubernetes 1.27.1

### PR DESCRIPTION
**Pull request by aojea**
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [x] Thanks for contributing!

```release-note
NONE
```

Run against kubernetes 1.27

### CODE REVIEW

File: .github/workflows/conformance-k8s-kind.yaml [hunk 1]
```diff
@@ -203,7 +203,7 @@ jobs:
           # LoadBalancer|GCE|ExternalIP : require a cloud provider, some of them are GCE specifics
           # Netpol|NetworkPolicy : network policies, demand significant resources and use to be slow, better to run in a different job
           # Aggregator : Flaky, https://github.com/cilium/cilium/issues/24622.
-          # same.port.number.but.different.protocols|HostPort : #9207
+          # same.port.number.but.different.protocols|HostPort|should.serve.endpoints.on.same.port.and.different.protocols : #9207
```
Comment (aojea) [inline .github/workflows/conformance-k8s-kind.yaml:205]: yeah, I should have been more careful and choose the same name 🤷

Comment (nebril) [reply .github/workflows/conformance-k8s-kind.yaml:205]: Should we add a note about this skip in #24622 ?

**Review by nebril — APPROVED** [review]

File: .github/workflows/conformance-k8s-kind.yaml (hunk 1 above)
Comment (aojea) [reply .github/workflows/conformance-k8s-kind.yaml:205]: it is an unrelated test, this is a new test we added to cover #9207 and it fails as expected

https://github.com/cilium/cilium/issues/24622 is a complex test that I actually managed to narrow down in https://github.com/kubernetes/kubernetes/issues/117443 and it was fixed recently, so we should wait for next minor release in kubernetes to disable it

Comment (aojea) [reply .github/workflows/conformance-k8s-kind.yaml:205]: hmm, I'm not sure is the same issue, but this specific test has some history, let's skip it for the moment

**Review by aanm — APPROVED** [review]
Thanks for the PR. Do we have an URL with a successful run? AFAIK these changes are not executed in Cilium because it was opened from a fork.

**Comment by aojea** [comment]
> Thanks for the PR. Do we have an URL with a successful run? AFAIK these changes are not executed in Cilium because it was opened from a fork.

The PR runs the job with the actual code, https://github.com/cilium/cilium/actions/runs/4882353953/jobs/8712328353?pr=25258

**Comment by aojea** [comment]
rebased

**Comment by aojea** [comment]
hmm, it failed this test

> 2023-05-07T22:27:44.5462356Z   [38;5;9m[FAIL][0m [0m[sig-apps] ReplicationController [38;5;9m[1m[It] should serve a basic image on each replica with a public image  [Conformance][0m

**Comment by aojea** [comment]
1/1 pass

**Comment by aojea** [comment]
2/2

**Comment by aojea** [comment]
This should be ready, 3/3 in a row green

**Comment by aojea** [comment]
run succesfull https://pipelines.actions.githubusercontent.com/serviceHosts/a38d642a-0bf8-46af-9d36-efb1e8318621/_apis/pipelines/1/runs/1475789/signedlogcontent/2?urlExpires=2023-05-17T10%3A54%3A13.0727759Z&urlSigningMethod=HMACV1&urlSignature=Uzwfih59ycziTjMLLfV9ven4PU4yIqazedfnlOJxPi4%3D

```
2023-05-17T08:16:36.3174568Z Ginkgo ran 1 suite in 20m8.976989507s
2023-05-17T08:16:36.3179816Z Test Suite Passed
2023-05-17T08:16:36.5267791Z ##[group]Run actions/upload-artifact@0b7f8abb1508181956e8e162db84b466c27e18ce
2023-05-17T08:16:36.5268400Z with:
2023-05-17T08:16:36.5272766Z   name: kubernetes-e2e-junit
2023-05-17T08:16:36.5273296Z   path: ./_artifacts/*.xml
2023-05-17T08:16:36.5294507Z   if-no-files-found: warn
2023-05-17T08:16:36.5295156Z env:
2023-05-17T08:16:36.5295459Z   kind_version: v0.19.0
2023-05-17T08:16:36.5295817Z   cluster_name: cilium-testing
2023-05-17T08:16:36.5296195Z   cilium_cli_version: v0.14.2
2023-05-17T08:16:36.5297120Z   cilium_cli_ci_version: 
2023-05-17T08:16:36.5297458Z   k8s_version: v1.27.1
2023-05-17T08:16:36.5298004Z   IP_FAMILY: ipv4
2023-05-17T08:16:36.5298741Z   QUAY_ORGANIZATION: cilium
2023-05-17T08:16:36.5299441Z   QUAY_ORGANIZATION_DEV: cilium
2023-05-17T08:16:36.5300078Z   QUAY_CHARTS_ORGANIZATION_DEV: cilium-charts-dev
```

**Review by aanm — APPROVED** [review]
Merging given that the only change on this PR had its test run passed successfully.

**Pull request merged**